File: models/notification_log.py
from datetime import datetime

from sqlalchemy import Column, DateTime, Integer, Text

from app.database import Base


class NotificationLog(Base):
    __tablename__ = "notification_logs"

    id = Column(Integer, primary_key=True, index=True)
    channel = Column(Text, nullable=False)
    title = Column(Text, nullable=False)
    content = Column(Text, nullable=False, default="")
    success = Column(Integer, nullable=False, default=0)
    response_message = Column(Text, nullable=False, default="")
    attempts = Column(Integer, nullable=False, default=1)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)

File: src/lib/notification-log.ts
import { prisma } from "@/lib/prisma";

export async function writeNotificationLog(params: {
  channel: string;
  title: string;
  content: string;
  success: boolean;
  responseMessage: string;
  attempts: number;
}): Promise<void> {
  const now = new Date();
  await prisma.notificationLog.create({
    data: {
      channel: params.channel,
      title: params.title,
      content: params.content.slice(0, 4000),
      success: params.success ? 1 : 0,
      responseMessage: params.responseMessage.slice(0, 500),
      attempts: params.attempts,
      createdAt: now,
    },
  });
}
